### PR TITLE
delete uncached files on manager disconnect

### PR DESCRIPTION
**Pull request by colinthomas-z80**
## Proposed changes

#3497 

## Post-change actions

Put an 'x' in the boxes that describe post-change actions that you have done.
The more 'x' ticked, the faster your changes are accepted by maintainers.

- [x] `make test`       Run local tests prior to pushing.
- [x] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [x] `make lint`       Run lint on source code prior to pushing.
- [x] Manual Update     Did you update the manual to reflect your changes, if appropriate? This action should be done after your changes are approved but not merged.
- [x] Type Labels       Select github labels for the type of this change: bug, enhancement, etc.
- [x] Product Labels    Select github labels for the product affected: TaskVine, Makeflow, etc.
- [x] PR RTM            Mark your PR as ready to merge.

### CODE REVIEW

**Review by dthain — CHANGES_REQUESTED** [review]
A couple things -
- This needs to happen when the worker exits and cleans up, but also when it starts up.  (because it could exit by crashing or being killed)
- The cache objects can be files as well as directories, so `unlink` is not enough.  Use `trash` to delete objects atomically.

**Comment by colinthomas-z80** [comment]
@dthain I'm not sure how the worker would start up using the same tmp cache directory as before, therefore it would not find the uncacheable files? Nevertheless this should provide that extra safeguard

**Comment by btovar** [comment]
@colinthomas-z80 Maybe if the --workdir is specified? (I don't remember if it is used directly, or a tmp dir is created in it.)

**Comment by dthain** [comment]
You are right, we do not currently start up in the same directory, and I had forgotten that.

This is mainly because we have not yet worked out how multiple concurrent workers could share the same cache directory safely.  But we are going in that direction, so I think it's good to cover the case...

File: taskvine/src/worker/vine_cache.c [hunk 1]
```diff
@@ -110,7 +111,14 @@ void vine_cache_load(struct vine_cache *c)
 			if (!strcmp(d->d_name, ".."))
 				continue;
 
-			debug(D_VINE, "found %s in cache", d->d_name);
+			if (strstr(d->d_name, "-meta-") || strstr(d->d_name, "-rnd-")) {
+				char *filepath = string_format("cache/%s", d->d_name);
```
Comment (dthain) [inline taskvine/src/worker/vine_cache.c:115]: Use `vine_cache_full_path` to get the right thing here.

**Review by dthain — CHANGES_REQUESTED** [review]

File: taskvine/src/worker/vine_cache.c [hunk 2]
```diff
@@ -110,7 +111,14 @@ void vine_cache_load(struct vine_cache *c)
...
+				debug(D_VINE, "found non-workflow cache %s, trashing file", d->d_name);
+				trash_file(filepath);
+				free(filepath);
```
[the remark this answers is not in the document]
Comment (colinthomas-z80) [reply taskvine/src/worker/vine_cache.c:118]: Probably something interesting

**Review by dthain — APPROVED** [review]

**Pull request merged**